run.cc

```diff
@@ -60,20 +61,84 @@ void runTests() {
     std::cout << (test() ? " ok" : " FAILED!") << std::endl;
 }
 
+
 int main() {
   using namespace std;
+  // Exercise 1c)
+  std::vector<Data> allSets;
+  std::string filenames[] = {"exp_A", "exp_B", "exp_C", "exp_D"};
 
   cout << "******************************************************" << endl;
+  cout << endl << "Exercise 1a)" << endl;
   runTests();
   cout << "******************************************************" << endl;
+  cout << endl << "Exercise 1b)" << endl;
   // create an object which holds data of experiment A
   Data datA("exp_A");
-
   // here is the data from experiment A
   cout << "bin 27: from " << datA.binLow(27) << " to " << datA.binHigh(27)
        << endl;
   cout << "measurement of experiment A in bin 27: " << datA.measurement(27)
        << endl;
+  // Exercise 1b)
+  cout << "uncertainty of experiment A in bin 27: " << datA.error(27)
+       << endl << endl;
+
+  // Exercise 1c)
+  cout << "******************************************************" << endl;
+  cout << endl << "Exercise 1c)" << endl;
+  for (int i = 0; i < 4; i++) {
+    Data set(filenames[i]);
+    allSets.push_back(set);
+    cout << "cros-section of experiment " << filenames[i]
+         << " in bin 27: " << set.measurement(27)
+         << " +- " << set.error(27) << endl;
+  }
+  
+  // Exercise 1d), 2a)
+  cout << "******************************************************" << endl;
+  cout << endl << "Exercise 1d), 2a)" << endl;
+  int d;
+  for (int i = 0; i <=2; i++) {
+    for (int j = i + 1; j <= 3; j++) {
+      cout << "Compatibility check of " << filenames[i] 
+           << " with " << filenames[j] << ", measurements differing"
+           << endl;
+      for (int n = 1; n <= 3; n++) {
+        d = allSets[i].checkCompatibility(allSets[j], n);
+        cout << "    more than " << n << " * sigma = " << d << " points" << endl;
+      }
+      cout << endl;
+    }
+  }
+
```

Comment:
In 2a) you didn't answer how many deviations you would typically expect even if the data were compatible. If you draw two samples from the same distribution, you would still expect some variation and even the probability to see a datapoint that has a deviation of 3 sigma is non-zero. So what you should have done here is just plug in the gaussian probability outside of 3 sigma (and 2 sigma) and multiply this with the number of datapoints, then see if the number of actual deviations agrees with what you computed.  Since your print out is correct, you'll still get 0.5 points for 2a)